distributed/client.py

```diff
@@ -680,24 +682,29 @@ def _repr_html_(self):
             info = False
             scheduler = self.scheduler
 
-        if scheduler is not None:
+        if scheduler is None:
             text = ("<h3>Client</h3>\n"
                     "<ul>\n"
-                    "  <li><b>Scheduler: </b>%s\n") % scheduler.address
+                    "  <li><b>Scheduler: not connected</b>\n")
         else:
             text = ("<h3>Client</h3>\n"
                     "<ul>\n"
-                    "  <li><b>Scheduler: not connected</b>\n")
-        if info and 'bokeh' in info['services']:
-            protocol, rest = scheduler.address.split('://')
-            port = info['services']['bokeh']
-            if protocol == 'inproc':
-                host = 'localhost'
-            else:
-                host = rest.split(':')[0]
-            template = dask.config.get('distributed.dashboard.link')
-            address = template.format(host=host, port=port, **os.environ)
-            text += "  <li><b>Dashboard: </b><a href='%(web)s' target='_blank'>%(web)s</a>\n" % {'web': address}
+                    "  <li><b>Scheduler: </b>%s\n") % scheduler.address
+
+            if info and 'bokeh' in info['services']:
+                urlparts = urlparse(scheduler.address)
+                if urlparts.scheme == 'inproc':
+                    host = 'localhost'
+                else:
+                    host = urlparts.hostname
+                port = info['services']['bokeh']
+                template = dask.config.get('distributed.dashboard.link')
+                ns = os.environ.copy()
+                ns['host'] = host
+                ns['port'] = port
```

Comment:
Since there are multiple ports in a distributed deployment perhaps this would be better renamed to `bokeh_port` to allow future usage of the `_local_config` without fear of name clashes?